src/dataworker/Dataworker.ts

```diff
@@ -13,40 +21,134 @@ export class Dataworker {
     readonly multiCallBundler: MultiCallBundler | any
   ) {}
 
-  async update() {
-    // Grab all deposits from SpokePoolClient
-    // Map deposits to their destination chain IDs
-    // Grab all fills from SpokePoolClient
-    // Filter out invalid fills for deposits
-    // Group remaining fills by destination chain ID
+  // Common data re-formatting logic shared across all data worker public functions.
+  _loadData(): { unfilledDeposits: UnfilledDeposits; fillsToRefund: FillsToRefund } {
+    // For each origin chain spoke pool client:
+    //     For all destination spoke pool client's:
+    //         Store deposits that are sent from origin chain to destination chain as an UnfilledDeposit
+    //             Associate this UnfilledDeposit with the destination chain and set fillAmountRemaining
+    //         Grab all fills on destination client
+    //             Attempt to map fill to an UnfilledDeposit sent from the origin chain
+    //             If a match is found, save fill and key by repaymentChainId => refundAddress
+
+    const unfilledDeposits: UnfilledDeposits = {};
+    const fillsToRefund: FillsToRefund = {};
+
+    for (const originChainId of Object.keys(this.spokePoolClients)) {
+      const originClient = this.spokePoolClients[originChainId];
+      if (!originClient.isUpdated())
+        throw new Error(`origin spokepoolclient with chain ID ${originChainId} not updated`);
```

Comment:
silly nit but let's try keep the total line length on these kinds of errors down. ideally it's just a one liner.

```suggestion
      if (!originClient.isUpdated()) throw new Error(`origin spokepoolclient on chain ${originChainId} not updated`);
```